.github/workflows/dotnet.yaml

```diff
@@ -141,3 +141,20 @@ jobs:
           name: nuget-package
       - name: Publish to NuGet
         run: dotnet nuget push "ApiSurface.*.nupkg" --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json
+
+  github-release:
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.repository.fork && github.ref == 'refs/heads/main' }}
+    needs: [all-required-checks-complete]
+    environment: release
+    permissions:
+      contents: write
```

Comment:
Seems a bit sad that GitHub's perms model isn't more finely grained here. This gives permission to update the commit hash of the `main` branch, for example.

Reply:
This job executes external code that you do not have control of: `actions/download-artifact@v4` and `ncipollo/release-action@v1`. Attacks on that code will be able to make arbitrary use of the `write` permission.

It is good practice to pin actions to a specific commit (not a tag like `v1` that can change over time) plus a thorough review of that commit's code.

Reply:
Dependabot supports those commit versions as well, it keeps suggesting latest versions (here commits): https://github.com/EnricoMi/publish-unit-test-result-action/pull/473

Reply:
We haven't reached a point where we should distrust the tags yet. If we did, a problem would be a lot bigger than what we are discussing here, and there would hundreds of PRs opened in this org alone to mitigate the issue. I do agree with the practice, but not so far to block PR from merging (i am aware it's approved). I will add though that when it comes to github actions (official and 3rd party), that we plan to have proxy actions in place of those, where we would do exactly what is mentioned (pin to a commit, and update to the next if review of the changes allows it). That will generate a lot of PRs over time as expected. I hope this gives some level of serenity

Reply:
Having just checked what that action does, a single line of curl would be simpler and easier to review :P

Reply:
True that 😅